### PR TITLE
Rubocop performance

### PR DESCRIPTION
**Pull request by deivid-rodriguez**
This is #564 rebased and with the feedback given there applied.

### CODE REVIEW

File: lib/sprockets/base.rb [hunk 1]
```diff
@@ -19,7 +19,7 @@ module Sprockets
 
   class DoubleLinkError < Sprockets::Error
     def initialize(parent_filename:, logical_path:, last_filename:, filename:)
-      message = String.new
+      message = +""
```
Comment (deivid-rodriguez) [inline lib/sprockets/base.rb]: In the docs it [says something about `String.new` not being 100% equivalent to `+""`](http://www.rubocop.org/en/latest/cops_performance/#performanceunfreezestring). Not sure if it applies here?

Comment (bdewater) [reply lib/sprockets/base.rb]: No. But beyond that there's really no reason to concatenate a string in this fashion here, instead do:

```ruby
message = <<~MSG
  Multiple files with the same output path cannot be linked (#{logical_path.inspect})
  In #{parent_filename.inspect} these files were linked:
    - #{last_filename}
    - #{filename}
MSG
```

Comment (deivid-rodriguez) [reply lib/sprockets/base.rb]: I even [spared a local variable](https://github.com/rails/sprockets/pull/594/commits/f1ce104144f6d22ce3cae7f56eeffb7e006fa859)! :smile: Thanks!!

**Comment by deivid-rodriguez** [comment]
Ping!

**Comment by schneems** [comment]
Thanks!

**Pull request merged**